docs/src/guide/discretisation.jl

```diff
@@ -198,22 +201,108 @@ Nrange = 10:10:100
 plot(Nrange, abs.(fconv.(Nrange) .- fconv(200)); yaxis=:log, legend=false)
 
 # ### Exercise 2
-# !!! note "TODO"
-#     This solution has not yet been written. Any help with a PR is appreciated.
+#- \langle e_G, e_{G'}\rangle = ∫_0^{2π} e_G^\ast(x) e_{G'}(x) d x = 1/2π ∫_0^{2π} e^i(G'-G)x d x$
+# Since `e^{iy}` is a periodic function with period $2\pi$, $\int_0^{2\pi} e^{i m y} = \delta_{0,m}$. Therefore:
+# if G≠G':
+#$\langle e_G, e_{G'}\rangle = 0$
+# if G=G':
+#$\langle e_G, e_{G'}\rangle = 1$
+# Therefore:
+#$\langle e_G, e_{G'}\rangle = δ_{G, G'}$
+#- Assuming $V(x) = \cos(x)$:
+#$\langle e_G, H e_{G'}\rangle = \frac 1 2 ∫_0^{2π} e_G^\ast(x) H e_{G'}(x) d x$
+# We start by applying the Hamiltonian on the corresponding function:
+#$H e_{G'}(x) = - \frac 1 2 (-|G|^2) \frac 1 {\sqrt{2π}} e^{iG'x) + cos(x) \frac 1 {\sqrt{2π}} e{iG'x}$
+# Then, using the previous result and the fact that $cos(x) = \frac 1 2 \left(e{ix} + e{-ix})$, we get:
+#$\langle e_G, H e_{G'}\rangle = \frac 1 2 G^2  δ_{G, G'} + \frac 1 {4π} \left(∫_0^{2π} (e^{ix})^{G'-G+1} d x + ∫_0^{2π} (e^{ix})^{G'-G-1} d x)$
+#$= \frac 1 2 \left(|G|^2 \delta_{G,G'} + \delta_{G, G'+1} + \delta_{G, G'-1}\right)$
+#- A more general $V(x)$ has to be periodic over $[0, 2\pi]$, therefore the complex eponential Fourier series can be used:
+#$V(x) = sum_{n=- \infty}^{\infty} v_n e{-inx}$
+# One can think of it as changing the basis of the potential function to the plane wave basis. Therefore :
+#$| v_n, e{iG'} \rangle = frac 1 {\sqrt{2π}} sum_{n=- \infty}^{\infty} v_n e^{i(G'-n)x}$
+#$\langle e_G, V e_{G'}\rangle = frac 1 {2π} sum_{n=- \infty}^{\infty} ∫_0^{2π} v_n e{i(G'-G-n)x} d x$
+#$= \frac 1 {2π} sum_{n=- \infty}^{\infty} v_n delta_{G, G'- n}$
+# Therefore :
+#$\langle e_G, H e_{G'}\rangle = \frac 1 2 |G|^2 \delta_{G,G'} + sum_{n=0}^{\infty} v_n delta_{G, G' \pm n}$
 #
 # ### Exercise 3
-# !!! note "TODO"
-#     This solution has not yet been written. Any help with a PR is appreciated.
+# The Hamiltonian matrix for the plane waves method can be found this way:
+## Plane waves Hamiltonian -½Δ + cos on [0, 2pi].
+
+using LinearAlgebra
+
+function build_plane_waves_matrix_cos(N::Integer)
+    # Plane wave approximation to -½Δ
```

Comment:
When you put in-line  comments in code segments, they need to start with ## ... Ask me later why if you are curious